File: src/main/java/com/demonwav/mcdev/exception/MinecraftSetupException.java
package com.demonwav.mcdev.exception;

import org.jetbrains.annotations.NotNull;

import javax.swing.JComponent;

public class MinecraftSetupException extends Exception {
    private JComponent j;

    public MinecraftSetupException(@NotNull String msg, @NotNull JComponent j) {
        super(msg);
        this.j = j;
    }

    public JComponent getJ() {
        return j;
    }

    public String getError() {
        switch (getMessage()) {
            case "empty":
                return "<html>Please fill in all required fields</html>";
            case "bad":
                return "<html>Please enter author and plugin names as a comma separated list</html>";
            case "fillAll":
                return "<html>Please fill in all fields</html>";
            default:
                return String.format("<html>%s</html>", getMessage());
        }
    }
}

File: src/main/kotlin/com/demonwav/mcdev/exception/MinecraftSetupException.kt
/*
 * Minecraft Dev for IntelliJ
 *
 * https://minecraftdev.org
 *
 * Copyright (c) 2017 minecraft-dev
 *
 * MIT License
 */

package com.demonwav.mcdev.exception

import com.demonwav.mcdev.asset.MCMessages
import javax.swing.JComponent

class MinecraftSetupException(msg: String, val j: JComponent) : Exception(msg) {

    val error: String
        get() {
            when (message) {
                "empty" -> return "<html>${MCMessages["setup.error.empty"]}</html>"
                "bad" -> return "<html>${MCMessages["setup.error.bad"]}</html>"
                "fillAll" -> return "<html>${MCMessages["setup.error.fill_all"]}</html>"
                else -> return "<html>$message</html>"
            }
        }
}
